### ISSUE TITLE
OAuth Login Loop on Spinnaker > 1.13.12 and Gate 1.7.2

### ISSUE THREAD

**Issue by Crazyigor1987** [spinnaker/spinnaker#5331]
### Issue Summary:
KeyCloak OAuth indefinitely redirected like an infinite loop

### Cloud Provider(s):
BareMetal with Kubernetes

### Environment:
Spinnaker 1.17.6 installed with Halyard to Kubernetes
Istio Ingress for spinnaker(Deck) and spinnaker-api(Gate)
KeyCloak for OpenID Connect

### Feature Area:
Authentication

### Description:
I've configured OAuth 2.0 Authentification with my custom provider KeyCloak. Just like described in the manual https://www.spinnaker.io/setup/security/authentication/oauth/ and the blog from Kasun Sameera https://medium.com/faun/spinnaker-authentication-with-keycloak-84db39f1123b . My login and the userinfomapping are valid and everything has worked on Spinnaker 1.13.12 with gate 1.7.2 

Now after an update to spinnaker 1.17.6, or in generell everything above 1.13.12, I got stuck in an infinite loop in the authentication page.  The Browser Console gives me this output in loop:

```
GEThttp://spinnaker-api.my.domain/auth/redirect?to=http%3A%2F%2Fspinnaker.my.domain%2F%23%2Fsearch
[HTTP/1.1 302 Found 49ms]

GEThttp://spinnaker-api.my.domain/login
[HTTP/1.1 302 Found 50ms]

GEThttps://oidc.my.domain/auth/realms/master/protocol/openid-connect/auth?client_id=spinnaker&redirect_uri=http://spinnaker-api.my.domain/login&response_type=code&scope=roles%20email%20profile&state=QK7Ttt
[HTTP/1.1 302 Found 84ms]

GEThttp://spinnaker-api.my.domain/login?state=QK7Ttt&session_state=3dcd6c0b-1dde-4a2b-993f-01d944fc2c55&code=8aa182c6-b9f0-49dd-a0db-5c266ec555d8.3dcd6c0b-1dde-4a2b-993f-01d944fc2c55.4a9c5d80-90b4-43bb-af38-d4535e8491d3
[HTTP/1.1 302 Found 55ms]

GEThttp://spinnaker-api.my.domain/login
[HTTP/1.1 302 Found 47ms]

GEThttps://oidc.my.domain/auth/realms/master/protocol/openid-connect/auth?client_id=spinnaker&redirect_uri=http://spinnaker-api.my.domain/login&response_type=code&scope=roles%20email%20profile&state=yE9EeL
[HTTP/1.1 302 Found 11ms]

GEThttp://spinnaker-api.my.domain/login?state=yE9EeL&session_state=3dcd6c0b-1dde-4a2b-993f-01d944fc2c55&code=cc7724a7-5b6a-481e-bbe8-7084982a8895.3dcd6c0b-1dde-4a2b-993f-01d944fc2c55.4a9c5d80-90b4-43bb-af38-d4535e8491d3
[HTTP/1.1 302 Found 51ms]

GEThttp://spinnaker-api.my.domain/login
[HTTP/1.1 302 Found 46ms]

GEThttps://oidc.my.domain/auth/realms/master/protocol/openid-connect/auth?client_id=spinnaker&redirect_uri=http://spinnaker-api.my.domain/login&response_type=code&scope=roles%20email%20profile&state=FGYePq
[HTTP/1.1 302 Found 11ms]

GEThttp://spinnaker-api.my.domain/login?state=FGYePq&session_state=3dcd6c0b-1dde-4a2b-993f-01d944fc2c55&code=2349a222-a838-4a77-94a7-2322d426cd90.3dcd6c0b-1dde-4a2b-993f-01d944fc2c55.4a9c5d80-90b4-43bb-af38-d4535e8491d3
[HTTP/1.1 302 Found 50ms]

GEThttp://spinnaker-api.my.domain/login
[HTTP/1.1 302 Found 45ms]

GEThttps://oidc.my.domain/auth/realms/master/protocol/openid-connect/auth?client_id=spinnaker&redirect_uri=http://spinnaker-api.my.domain/login&response_type=code&scope=roles%20email%20profile&state=QWfR4d
[HTTP/1.1 302 Found 13ms]

GEThttp://spinnaker-api.my.domain/login?state=QWfR4d&session_state=3dcd6c0b-1dde-4a2b-993f-01d944fc2c55&code=668c295b-66be-4c6d-8400-8ee243deb3b5.3dcd6c0b-1dde-4a2b-993f-01d944fc2c55.4a9c5d80-90b4-43bb-af38-d4535e8491d3
[HTTP/1.1 302 Found 51ms]
```

### Steps to Reproduce:

- Install latest Spinnaker with halyard
- Create valid KeyCloak OIDC Endpoints with Users
- Create ~/.hal/default/profiles/gate-local.yml with these settings
```
security:
  oauth2:
    enabled: true
    client:
      clientId: ###CLIENTID###
      clientSecret: ###CLIENTSECRET###
      userAuthorizationUri: ###userAuthorizationUri###
      accessTokenUri: ###accessTokenUri###
      scope: roles,email,profile
    resource:
      userInfoUri: ###userInfoUri###
    userInfoMapping:
      email: email
      firstName: given_name
      lastName: family_name
      username: preferred_username

```
And deploy it

hal config security authn oauth2 enable
hal deploy apply

### Additional Details:

Might be a regression. Works like a charm in Spinnaker 1.13.12

**Comment by Crazyigor1987** [comment]
I was able to indicate the problem. I've used my Keycloak Server outside my K8s Cluster. The Spinnaker Gate was not able to perform an DNS Resolve of my Domain in his Alpine Variant because of this Bug: https://github.com/gliderlabs/docker-alpine/issues/476

It should be fixed in alpine 3.11.3.

My Workaround is to use spinnaker with ubuntu images.
`
hal config deploy edit --image-variant ubuntu
`

With these images i can deploy the latest Spinnaker and perform oauth2 validation.
